core/shared/src/main/scala/zio/internal/ConcurrentWeakHashSet.scala

```diff
@@ -140,17 +144,20 @@ private[zio] class ConcurrentWeakHashSet[V](
    *   approximate number of threads that will access the set concurrently
    *   (default [[ConcurrentWeakHashSet.DefaultConcurrencyLevel]])
    */
-  protected def concurrencyLevel: Int =
-    ConcurrentWeakHashSet.DefaultConcurrencyLevel
+  def concurrencyLevel: Int = ConcurrentWeakHashSet.DefaultConcurrencyLevel
 
   /**
    * Enforce cleanup of dead references in the set. By default cleanup is
    * performed automatically when the set is modified, but some operations (like
    * e.g. `size`) do not modify the set and therefore do not trigger cleanup.
    */
   def gc(): Unit = {
-    val iterator = this.segments.iterator
-    while (iterator.hasNext) iterator.next().restructureIfNecessary(false)
+    var idx    = 0
+    val length = this.segments.length
+    while (idx < length) {
+      this.segments(idx).restructureIfNecessary(false)
+      idx += 1
+    }
```

Comment:
I'm not sure if it's better to use a mutable index like this than using an iterator